src/event.js

```diff
@@ -38,17 +39,34 @@ jQuery.event.remove = function( elem, types, handler, selector, mappedTypes ){
 	eventRemove.call( this, elem, hoverHack( types ) || "", handler, selector, mappedTypes );
 };
 
-jQuery.fn.error = function() {
-	var args = Array.prototype.slice.call( arguments, 0);
-	migrateWarn("jQuery.fn.error() is deprecated");
-	args.splice( 0, 0, "error" );
-	if ( arguments.length ) {
-		return this.bind.apply( this, args );
-	}
-	// error event should not bubble to window, although it does pre-1.7
-	this.triggerHandler.apply( this, args );
-	return this;
-};
+jQuery.each( [ "load", "unload", "error" ], function( _, name ) {
+
+	jQuery.fn[ name ] = function() {
+		var args = Array.prototype.slice.call( arguments, 0 );
+		migrateWarn( "jQuery.fn." + name + "() is deprecated" );
+
+		// If this is an ajax load() the first arg should be the string URL;
+		// technically this could also be the "Anything" arg of the event .load()
+		// which just goes to show why this dumb signature has been deprecated!
+		// jQuery custom builds that exclude the Ajax module justifiably die here.
+		if ( name === "load" && typeof arguments[ 0 ] === "string" ) {
+			return oldLoad.apply( this, arguments );
+		}
+
+		args.splice( 0, 0, name );
+		if ( arguments.length ) {
+			return ( this.bind || this.on ).apply( this, args );
```

Comment:
Good point. It was a spur-of-the-moment thought that if someone used a custom build that excluded `.bind()` it would be nice if this still worked. It's worth thinking about whether we want to support those kind of builds though, because if we did all those cases like the one you pointed out should be changed as well.